EmployeeLeaveTracking/EmployeeLeaveTracking.Services/Services/LeaveRequestService.cs

```diff
@@ -202,10 +202,45 @@ public NewLeaveRequestDTO NewCreateNewLeaveRequest(NewLeaveRequestDTO leaveReque
                 IsViewed = false
             });
 
-
             _dbContext.Notifications.Add(notification);
             _dbContext.SaveChanges();
 
+            // Retrieve employee and manager details
+            User employee = _dbContext.Users.FirstOrDefault(u => u.Id == leaveRequest.EmployeeId);
+            User manager = _dbContext.Users.FirstOrDefault(u => u.Id == leaveRequest.ManagerId);
+
+            if (employee == null || manager == null)
+            {
+                throw new Exception("Employee or manager not found.");
+            }
+
+            User managerEmail = _dbContext.Users.FirstOrDefault(u => u.Id == leaveRequest.ManagerId);
+
+            if (managerEmail == null)
+            {
+                throw new Exception("Manager not found.");
+            }
+
+            //accessing the resource file
+            ResourceManager resourceManager = new ResourceManager("EmailService.EmailTemplates", typeof(EmailHelper).Assembly);
+
+            //retrieving the LeaveRequestTemplate from the resource file
+            string leaveRequestTemplate = resourceManager.GetString("LeaveRequestTemplate");
+
+            //replacing placeholders with appropriate values
+            string emailBody = string.Format(leaveRequestTemplate,
+                employee.FirstName + " " + employee.LastName,
+                manager.FirstName + " " + manager.LastName,
+                leaveRequest.StartDate.ToShortDateString(),
+                leaveRequest.EndDate.ToShortDateString(),
+                leaveRequest.TotalDays,
+                leaveRequest.RequestComments);
+
+            EmailHelper emailService = new EmailHelper();
+            string emailToAddress = manager.Email; //manager's email address
+            string subject = "Leave Request";
```

Comment:
This can be moved to constants/resources

Reply:
Added Message from Constant File